ui/src/renderer/ComponentProxy.vue

```diff
@@ -310,11 +311,25 @@ export default {
 			const vnodeProps = {
 				...getRootElProps(),
 			};
-			const renderedComponent = h(template, vnodeProps, {
-				default: defaultSlotFn,
-			});
 
-			return renderedComponent;
+			const parentId = instancePath.at(-2)?.componentId;
+			const allowedTypes = !parentId
+				? ["root"]
```

Comment:
Logic looks good, but not the most readable, can you refactor a bit? 

`if (isParentSuitable()) {
  return render normally...
}

return getRenderErrorVNode(type, "parent not suitable");
`